arcon_state/src/backend/mod.rs

```diff
@@ -124,16 +124,18 @@ pub trait Backend:
                     epoch = epoch
                 ));
 
-                Self::restore(&state_path, &latest_checkpoint_path)
+                Self::restore(&state_path, &latest_checkpoint_path, id)
             }
-            None => Self::create(&state_path),
+            None => Self::create(&state_path, id),
         }
     }
 
-    fn create(live_path: &Path) -> Result<Self>
+    fn return_name(&self) -> String;
```

Comment:
Let's use string literals instead and avoid cloning everywhere.

And no need to have return in the function name.

```rust
fn name(&self) -> &str;
```